ProcessMaker/processmaker
# Add php-cs-fixer and pre-commit hook installer

## Issue & Reproduction Steps
As part of the Laravel upgrade process, we will be formatting all PHP code to follow code standards rules. We need a way to enforce these standards going forward after the initial fix for all files.

## Solution
- This PR applies code standards formatting to all files.
- This PR adds a pre-commit git hook to enforce the new rules. If a files has formatting issues, committing will be prevented.

Packges should not need any updates other than .gitignore and the initial formatting. So far I only added this to package-webentry until I get feedback

## How to Test
1. In Core, run `composer install`. This should install the git pre-commit hooks.
1. Edit a php file and add some formatting changes. For example, change the indenting somewhere
1. Attempt to commit the changes. You should see an error.
1. Install the `feature/FOUR-6588` branch of package-webentry. This branch has the formatting changes already applied
1. Edit a php file in the web entry package to change formatting.
1. Attempt to commit the change. You should see an error.

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-6588
- https://github.com/ProcessMaker/package-webentry/tree/feature/FOUR-6588

## Code Review Checklist
